pkg/database/resync.go

```diff
@@ -4,45 +4,179 @@ package database
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"strings"
+	"sync"
+	"time"
 
 	"github.com/stolostron/search-indexer/pkg/metrics"
 	"github.com/stolostron/search-indexer/pkg/model"
 	"k8s.io/klog/v2"
 )
 
-// Overrides the existing state of a cluster with the new data.
-// NOTE: This logic is not optimized. We use the simplest approach because this is a failsafe to
-//       recover from rare sync problems. At the moment this is good enough without adding complexity.
+func logStepDuration(timer *time.Time, cluster, message string) {
+	klog.V(2).Infof("\t> %6s\t - [%10s] %s", time.Since(*timer).Round(time.Millisecond), cluster, message)
+	*timer = time.Now()
+}
+
+// Reset data for the to the incoming state.
 func (dao *DAO) ResyncData(ctx context.Context, event model.SyncEvent,
 	clusterName string, syncResponse *model.SyncResponse) error {
 
-	defer metrics.SlowLog(fmt.Sprintf("Slow Resync from cluster %s", clusterName), 0)()
+	defer metrics.SlowLog(fmt.Sprintf("Slow resync from cluster %s", clusterName), 0)()
 	klog.Infof(
-		"Starting Resync of cluster %s. This is normal, but it could be a problem if it happens often.",
-		clusterName)
+		"Starting resync of [%10s]. This is normal, but it could be a problem if it happens often.", clusterName)
+
+	wg := &sync.WaitGroup{}
+	wg.Add(2)
+	// Reset resources
+	go dao.resyncResources(ctx, wg, event.AddResources, clusterName, syncResponse)
+	// Reset edges
+	go dao.resyncEdges(ctx, wg, event.AddEdges, clusterName, syncResponse)
+	wg.Wait()
+
+	// TODO: Need to capture errors from the goroutines above.
+
+	klog.V(1).Infof("Completed resync of cluster %s", clusterName)
```

Comment:
Do you think it is helpful to add requestid to the info messages in this PR so that it is easy to correlate with collector while debugging?